sdk/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/Hash.scala

```diff
@@ -235,7 +235,7 @@ object Hash {
         domainId: String,
         ledgerEffectiveTime: Option[Time.Timestamp],
         submissionTime: Time.Timestamp,
-        submissionSeed: Hash,
+        nodeSeeds: SortedMap[NodeId, Hash],
```

Comment:
Not very comfortable with sorting according NodeId. 
It may be better to add the hash in the corresponding Node as you hashing them.

Reply:
If you prefer to hash the seeds independently, you can better sorted them in traversal with something like:
```

  val nodeSeeds: Map[NodeId, crypto.Hash] = ???
  val tx: VersionedTransaction = ???

  tx.fold(Vector.empty[crypto.Hash]) { 
    case (acc, (nid, _ : Node.Create | _: Node.Exercise)) => acc :+ nodeSeeds(nid)
    case (acc, _) => acc
  }
```

Reply:
Ah but you're right, I already did add the seed in the corresponding node, so we don't actually need to re-hash it in the metadata in the end no?

Reply:
Yes, it should be fine. 
Though there is a bug as your Node seed in you trace are Empty.

Reply:
Hmm no they're here ? https://github.com/digital-asset/daml/pull/20218/files#diff-82f760ccebcd2da5112b9d802093308a5fbac037b0161e390430485e38dedf76R78-R80